### PR TITLE
rubberband 1.8.1 (new formula)

### PR DESCRIPTION
**Pull request by david0**
Originally from http://tuohela.net/irc/rubberband.rb
Modified by mtolly to include libsndfile dependency,
and to install pkg-config file
plus tests, updates and fixes from myself

### CODE REVIEW

File: Library/Formula/rubberband.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Originally from http://tuohela.net/irc/rubberband.rb
+# Modified by mtolly to include libsndfile dependency,
+# and to install pkg-config file.
```
Comment (mikemcquaid) [inline Library/Formula/rubberband.rb]: You can remove these comments.

File: Library/Formula/rubberband.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  depends_on "pkg-config" => :build
+  depends_on "libsamplerate"
+  depends_on "libsndfile"
+
+  def install
+    system "make", "-f", "Makefile.osx"
+    bin.install "bin/rubberband"
+    lib.install "lib/librubberband.dylib"
+    include.install "rubberband"
+    (lib/"pkgconfig/rubberband.pc").write pc_file
```
Comment (mikemcquaid) [inline Library/Formula/rubberband.rb]: I don't think we should include this if upstream doesn't provide it.

File: Library/Formula/rubberband.rb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  depends_on "pkg-config" => :build
+  depends_on "libsamplerate"
+  depends_on "libsndfile"
+
+  def install
+    system "make", "-f", "Makefile.osx"
+    bin.install "bin/rubberband"
+    lib.install "lib/librubberband.dylib"
+    include.install "rubberband"
```
Comment (mikemcquaid) [inline Library/Formula/rubberband.rb]: No `make install` we can use?

File: Library/Formula/rubberband.rb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    Version: #{version}
+    Description:
+    Libs: -L${libdir} -lrubberband
+    Cflags: -I${includedir}
+    EOS
+  end
+
+  test do
+    assert_match /Pass 2: Processing.../, shell_output("rubberband -t2 " + test_fixtures("test.wav") + " out.wav 2>&1")
+    rm "out.wav"
```
Comment (mikemcquaid) [inline Library/Formula/rubberband.rb]: don't need to remove this, it's cleaned up automatically.

File: Library/Formula/rubberband.rb [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    Name: rubberband
+    Version: #{version}
+    Description:
+    Libs: -L${libdir} -lrubberband
+    Cflags: -I${includedir}
+    EOS
+  end
+
+  test do
+    assert_match /Pass 2: Processing.../, shell_output("rubberband -t2 " + test_fixtures("test.wav") + " out.wav 2>&1")
```
Comment (mikemcquaid) [inline Library/Formula/rubberband.rb]: Use `rubberband -t2 #{test_fixtures("test.wav")} out.wav` and don't redirect stderr if you can and instead match on some `stdout`.

File: Library/Formula/rubberband.rb (hunk 2 above)
Comment (david0) [inline Library/Formula/rubberband.rb]: It looks like it is provided for linux (autoconf), but not using the OS X makefile. 
But now that I think of it, I could use the included rubberband.pc.in and post-process that. Would that be ok?

File: Library/Formula/rubberband.rb (hunk 3 above)
Comment (david0) [inline Library/Formula/rubberband.rb]: No, there is one for linux (autoconf) but none in `Makefile.osx` unfortunately

File: Library/Formula/rubberband.rb (hunk 5 above)
Comment (david0) [inline Library/Formula/rubberband.rb]: all the output is on stderr, thats why im redirecting to match it.

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**